tests/main/snap-set/task.yaml

```diff
@@ -1,5 +1,30 @@
 summary: Check that `snap set` runs configure hook.
 
+details: |
+    The `snap set` command modifies configuration options for snaps. As part of
+    this process, configure hooks defined in the snap are run. This test checks
+    the following scenarios related to `snap set`:
+
+    1. If a snap has no configure hook defined, `snap set` fails.
+
+    2. If a configure hook errors, `snap set` fails.
+
+    3. Configuration options set via `snap set` can be retrieved by the
+    resulting configure hook.
+
+    4. Configuration options can be set to `null`, which removes the option.
+
+    5. Configuration options can be unset, which removes the option.
+
+    6. Setting an invalid key results in an error.
```

Comment:
It is unclear to me where the valid keys are defined. In the the test `snapctl-hooks` snap, for example, the only thing special there seems to be about the `command` key is that it is checked as follows:
```
command=$(snapctl get command)
case $command in
    ...
```
Is the distinction that there must be a variable with a name matching the key defined at some point in the configure hook?